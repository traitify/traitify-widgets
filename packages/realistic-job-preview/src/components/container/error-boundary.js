import PropTypes from "prop-types";
import {useState} from "react";
import useWidgetContext from "lib/hooks/use-widget-context";
import Boundary from "traitify/components/common/error-boundary";

function ErrorBoundary({children}) {
  const {dispatch} = useWidgetContext();
  const [count, setCount] = useState(0);
  const onError = (error) => {
    console.error(error); // eslint-disable-line no-console
    dispatch({error, type: "error"});
    setCount((total) => total + 1);
  };

  if(count > 2) { return null; }

  return <Boundary onError={onError}>{children}</Boundary>;
}

ErrorBoundary.propTypes = {children: PropTypes.node.isRequired};

export default ErrorBoundary;
