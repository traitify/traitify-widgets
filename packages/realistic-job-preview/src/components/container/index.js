import PropTypes from "prop-types";
import Context from "lib/context";
import ErrorBoundary from "./error-boundary";
import Theme from "./theme";
import useReducer from "./use-reducer";

function Container({children, className = null, ...props}) {
  const state = useReducer({...props});

  return (
    <Context.Provider value={state}>
      <Theme className={className}>
        <ErrorBoundary>{children}</ErrorBoundary>
      </Theme>
    </Context.Provider>
  );
}

Container.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};

export default Container;
