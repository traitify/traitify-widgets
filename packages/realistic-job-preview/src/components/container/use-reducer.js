import {useReducer} from "react";
import slice from "traitify/lib/common/object/slice";
import useDidUpdate from "traitify/lib/hooks/use-did-update";
import Cache from "traitify/lib/cache";
import Http from "traitify/lib/http";
import I18n from "traitify/lib/i18n";
import Listener from "traitify/lib/listener";

const setInitialState = ({assessmentID, cache, http, i18n, listener, locale, options}) => ({
  assessmentID,
  cache: cache || new Cache({namespace: "rjp"}),
  http: http || new Http(slice(options, ["authKey", "host", "version"])),
  i18n: i18n || new I18n(),
  listener: listener || new Listener(),
  locale: locale || "en-us",
  options: options || {}
});

const reducer = (state, action) => {
  switch(action.type) {
    case "error":
      return {...state, error: action.error};
    case "updateAssessmentID":
      return {...state, assessmentID: action.assessmentID};
    case "updateCache":
      return {...state, assessmentID: action.cache};
    case "updateHttp":
      return {
        ...state,
        http: action.http || new Http(slice(action.options, ["authKey", "host", "version"]))
      };
    case "updateI18n":
      return {...state, i18n: action.i18n || new I18n()};
    case "updateListener":
      return {...state, listener: action.listener || new Listener()};
    case "updateLocale":
      return {...state, locale: action.locale};
    case "updateOptions":
      return {...state, options: action.options};
    default:
      throw new Error("Invalid Action");
  }
};

export default function useStateReducer({
  assessmentID,
  http,
  i18n,
  listener,
  locale,
  options
}) {
  const [state, dispatch] = useReducer(
    reducer,
    {assessmentID, http, i18n, listener, locale, options},
    setInitialState
  );

  useDidUpdate(() => {
    dispatch({assessmentID, type: "updateAssessmentID"});
  }, [assessmentID]);

  useDidUpdate(() => {
    dispatch({http, options, type: "updateHttp"});
  }, [http]);

  useDidUpdate(() => {
    dispatch({i18n, type: "updateI18n"});
  }, [i18n]);

  useDidUpdate(() => {
    dispatch({listener, type: "updateListener"});
  }, [listener]);

  useDidUpdate(() => {
    dispatch({locale, type: "updateLocale"});
  }, [locale]);

  useDidUpdate(() => {
    dispatch({options, type: "updateOptions"});
  }, [options]);

  return {dispatch, ...state};
}
