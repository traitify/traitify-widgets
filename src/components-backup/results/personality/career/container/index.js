import PropTypes from "prop-types";
import TraitifyPropTypes from "lib/helpers/prop-types";
import useDidMount from "lib/hooks/use-did-mount";
import useDidUpdate from "lib/hooks/use-did-update";
import withTraitify from "lib/with-traitify";
import CareerFilter from "../filter";
import CareerList from "../list";
import CareerModal from "../modal";
import style from "./style.scss";

function CareerContainer(props) {
  const {ui, isReady, setElement} = props;
  const state = {};

  useDidMount(() => { ui.trigger("Careers.initialized", {props, state}); });
  useDidUpdate(() => { ui.trigger("Careers.updated", {props, state}); });

  return (
    isReady("results") && (
      <div className={style.container} ref={setElement}>
        <CareerFilter {...props} />
        <CareerList {...props} />
        <CareerModal {...props} />
      </div>
    )
  );
}
CareerContainer.propTypes = {
  isReady: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired,
  setElement: PropTypes.func.isRequired
};

export {CareerContainer as Component};
export default withTraitify(CareerContainer);
