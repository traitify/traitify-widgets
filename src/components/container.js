/* eslint-disable react/forbid-prop-types */
import PropTypes from "prop-types";
import {useEffect} from "react";
import {RecoilRoot, useRecoilState, useSetRecoilState} from "recoil";
import slice from "lib/common/object/slice";
import Http from "lib/http";
import I18n from "lib/i18n";
import Listener from "lib/listener";
import {
  assessmentIDState,
  benchmarkIDState,
  httpState,
  i18nState,
  listenerState,
  loadingState,
  localeState,
  optionsState,
  profileIDState
} from "lib/recoil";

function State({children, http: _http, i18n: _i18n, listener: _listener, options}) {
  const setAssessmentID = useSetRecoilState(assessmentIDState);
  const setBenchmarkID = useSetRecoilState(benchmarkIDState);
  const setHttp = useSetRecoilState(httpState);
  const setI18n = useSetRecoilState(i18nState);
  const setListener = useSetRecoilState(listenerState);
  const setLocale = useSetRecoilState(localeState);
  const setOptions = useSetRecoilState(optionsState);
  const setProfileID = useSetRecoilState(profileIDState);
  const [loading, setLoading] = useRecoilState(loadingState);
  const {
    assessmentID,
    benchmarkID,
    locale,
    profileID,
    ...extraOptions
  } = options;

  useEffect(() => {
    setOptions(extraOptions);
  }, [options]);

  useEffect(() => {
    if(!locale) { return; }

    setLocale(locale);
  }, [locale]);

  useEffect(() => {
    const http = _http || new Http(slice(options, ["authKey", "host", "version"]));

    setHttp(http);
  }, [_http]);

  useEffect(() => {
    const i18n = _i18n || new I18n();

    setI18n(i18n);
  }, [_i18n]);

  useEffect(() => {
    const listener = _listener || new Listener();

    setListener(listener);
  }, [_listener]);

  useEffect(() => {
    setAssessmentID(assessmentID);
  }, [assessmentID]);

  useEffect(() => {
    setBenchmarkID(benchmarkID);
  }, [benchmarkID]);

  useEffect(() => {
    setProfileID(profileID);
  }, [profileID]);

  useEffect(() => {
    setLoading(false);
  }, []);

  if(loading) { return null; }

  return children;
}

State.defaultProps = {
  http: null,
  i18n: null,
  listener: null,
  options: {}
};
State.propTypes = {
  children: PropTypes.node.isRequired,
  http: PropTypes.object,
  i18n: PropTypes.object,
  listener: PropTypes.object,
  options: PropTypes.shape({
    assessmentID: PropTypes.string,
    benchmarkID: PropTypes.string,
    locale: PropTypes.string,
    profileID: PropTypes.string
  })
};

export default function Container(props) {
  return (
    <RecoilRoot>
      <State {...props} />
    </RecoilRoot>
  );
}
