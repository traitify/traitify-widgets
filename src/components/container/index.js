import {RecoilRoot} from "recoil";
import ErrorBoundary from "./error-boundary";
import State from "./state";

export default function Container(props) {
  return (
    <RecoilRoot>
      <ErrorBoundary>
        <State {...props} />
      </ErrorBoundary>
    </RecoilRoot>
  );
}
