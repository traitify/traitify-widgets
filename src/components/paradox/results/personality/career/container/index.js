import PropTypes from "prop-types";
import {useEffect} from "react";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import CareerFilter from "../filter";
import CareerModal from "../modal";
import CareerResults from "../list";
import style from "./style.scss";

function Careers(props) {
  const {ui, isReady} = props;

  useEffect(() => {
    ui.trigger("Careers.initialized", {props});
  }, []);
  useEffect(() => {
    ui.trigger("Careers.updated", {props});
  });

  return (
    isReady("results") && (
      <div className={style.container}>
        <CareerFilter {...props} />
        <CareerResults {...props} />
        <CareerModal {...props} />
      </div>
    )
  );
}
Careers.propTypes = {
  isReady: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};
export {Careers as Component};
export default withTraitify(Careers);
