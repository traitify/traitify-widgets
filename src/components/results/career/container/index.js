import useComponentEvents from "lib/hooks/use-component-events";
import useResults from "lib/hooks/use-results";
import style from "./style.scss";
import CareerFilter from "../filter";
import CareerList from "../list";
import CareerModal from "../modal";

export default function CareerContainer() {
  const results = useResults({surveyType: "personality"});

  useComponentEvents("Careers");

  if(!results) { return null; }

  return (
    <div className={style.container}>
      <CareerFilter />
      <CareerList />
      <CareerModal />
    </div>
  );
}
