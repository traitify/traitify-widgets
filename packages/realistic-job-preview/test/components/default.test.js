import mutable from "traitify/lib/common/object/mutable";
import Component from "components/default";
import ComponentHandler from "support/component-handler";
import {mockAssessment} from "support/container/http";
import _assessment from "support/data/assessment.json";
import useContainer from "support/hooks/use-container";

jest.mock("components/results", () => (() => <div className="mock">Results</div>));
jest.mock("components/survey", () => (() => <div className="mock">Survey</div>));

describe("Default", () => {
  let assessment;
  let component;

  useContainer();

  beforeEach(() => {
    assessment = mutable(_assessment);
    assessment.completedAt = assessment.updatedAt;

    mockAssessment(assessment);
  });

  it("renders nothing", async() => {
    container.assessmentID = null;
    mockAssessment(null);
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders results", async() => {
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders survey", async() => {
    assessment.completedAt = null;
    mockAssessment(assessment);
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });
});
