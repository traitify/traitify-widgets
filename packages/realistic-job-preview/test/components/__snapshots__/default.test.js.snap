// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default renders nothing 1`] = `
<div
  className="rjp--container rjp--color-scheme-light"
/>
`;

exports[`Default renders results 1`] = `
<div
  className="rjp--container rjp--color-scheme-light"
>
  <div
    className="mock"
  >
    Results
  </div>
</div>
`;

exports[`Default renders survey 1`] = `
<div
  className="rjp--container rjp--color-scheme-light"
>
  <div
    className="mock"
  >
    Survey
  </div>
</div>
`;
